content/topics/java-specific/functional-interface/_index.md

```diff
@@ -0,0 +1,68 @@
+---
+content_type: topic
+ready: true
+tags:
+- functions
+title: Java Lambda expressions and Functional Interface
+---
+
+#### What are Lambda expressions:
+
+Introduced in Java 8 Lambda expressions are anonymous(don't have names) shortcode blocks that can take parameters and return a value just like methods.
+
+#### What are Functional Interface
+
+Functional Interface in Java is also called Single Abstract Method (SAM) interface. They can have many default and static methods but only one abstract method - this is why they are called SAM. They make it possible to create abstractions that multiple classes can use without copying and pasting code. This is especially helpful when developers need to create a complex abstraction with various methods and behaviors.
+
+All together Functional interface with lambda expressions help us write smaller and cleaner code by removing a lot of boilerplate code.
```

Comment:
Rather:

"Functional interfaces with lambda expressions help us to remove boilerplate code, this makes our code smaller, cleaner and faster to write"

If you read it aloud you'll get a feel for where punctuation should be. And sometimes you can flat out remove words that serve no purpose.